WordPress/Classes/ViewRelated/Stats/Shared Views/Date Chooser/SiteStatsTableHeaderView.swift

```diff
@@ -4,6 +4,10 @@ protocol SiteStatsTableHeaderDelegate: class {
     func dateChangedTo(_ newDate: Date?)
 }
 
+protocol SiteStatsTableHeaderDateButtonDelegate: SiteStatsTableHeaderDelegate {
```

Comment:
> As soon the same logic to get the start date from the intervals list is applied to the Stats detail and Post detail

I'm assuming one of these protocols will be removed when :point_up:  is done? And the accompanying (now defunct) `dateChangedTo` method in `SiteStatsPeriodTableViewController`?

Reply:
Oh, nevermind. I see what's happening. Sorry!

Reply:
Exactly, after the overview is done the idea is to start with the post chart and stats detail and use the new protocol, moving the logic from the header to the view model. After that the other delegate and `func calculateEndDate(from currentDate: Date, offsetBy count: Int = 1, unit: StatsPeriodUnit) -> Date?` can be removed.